src/ddx.h

```diff
@@ -75,27 +93,19 @@ void ddx_scaled_ylm(const void* ddx, int lmax, const double* x, int sphere, doub
  *                         be quite verbose for larger runs and should only be used for
  *                         debugging.
  * \param logfile          The logfile name (empty if no log)
+ * \param error            The ddX error object
  */
 void* ddx_allocate_model(int model, int enable_force, double solvent_epsilon,
                          double solvent_kappa, double eta, double shift, int lmax,
                          int n_lebedev, int incore, int maxiter, int jacobi_n_diis,
                          int enable_fmm, int fmm_multipole_lmax, int fmm_local_lmax,
                          int n_proc, int n_spheres, const double* sphere_centres,
                          const double* sphere_radii, int length_logfile,
-                         const char* logfile);
+                         const char* logfile, const void* error);
```

Comment:
Strictly speaking the `const` qualification is wrong here, because you may modify the `error` object (may change the error flag).

Reply:
same in most other places where the error pointer pops up.

Reply:
Right, I made some modifications.

Now the error is passed as `const void* error` only to the functions `ddx_get_error_flag` and `ddx_get_error_message`, which do not modify the error flag. Everywhere else it is passed as "void* error".